core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -186,13 +186,13 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     try {
       val statusList = Option(fs.listStatus(new Path(logDir))).map(_.toSeq)
         .getOrElse(Seq[FileStatus]())
-      var newLastModifiedTime = lastModifiedTime
       val logInfos: Seq[FileStatus] = statusList
         .filter { entry =>
           try {
             getModificationTime(entry).map { time =>
-              newLastModifiedTime = math.max(newLastModifiedTime, time)
-              time >= lastModifiedTime
+              val fileName = entry.getPath.getName
+              val oldLastModifiedTime = lastModifiedTimes.getOrElse(fileName, -1L)
+              time > oldLastModifiedTime
```

Comment:
If the app is in process, it doesn't matter if we don't reprocess the log file. After the app completed, its' file name will change, and it will get the `oldLastModifiedTime = -1', so `time > oldLastModifiedTime` will be `true`, and the log file will reprocess.